frame/staking/src/lib.rs

```diff
@@ -1268,6 +1378,93 @@ impl<T: Trait> Module<T> {
 
 	// MUTABLES (DANGEROUS)
 
+	fn do_payout_nominator(who: T::AccountId, era: EraIndex, validators: Vec<T::AccountId>)
+		-> DispatchResult
+	{
+		let mut nominator_ledger = <Ledger<T>>::get(&who).ok_or_else(|| Error::<T>::NotController)?;
+		if nominator_ledger.next_reward > era {
+			return Err(Error::<T>::InvalidEraToReward.into());
+		}
+
+		nominator_ledger.next_reward = era + 1;
+		<Ledger<T>>::insert(&who, &nominator_ledger);
+
+		let mut reward = Perbill::zero();
+		let era_reward_points = <ErasRewardPoints<T>>::get(&era);
+		for validator in validators {
+			let commission = Self::eras_validator_prefs(&era, &validator).commission;
+			let validator_exposure = <ErasStakers<T>>::get(&era, &validator);
```

Comment:
currently unbounded (phragmen can give a validator any number of nominators). that will need to be fixed.

Reply:
note that if the unbounded issue is only for nominator reward we can store this information also per-nominator so it doesn't query this unbounded exposure for each validator but only one structure which holds all its information.

Reply:
This is fixed by introducing ErasStakersClipped which hold only the N biggest nominators for each stakers, such clipped exposure is only used for reward.